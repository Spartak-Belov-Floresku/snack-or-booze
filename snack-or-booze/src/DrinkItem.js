import { useState, useEffect } from "react";
import { Redirect, useParams, Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";
import SnackOrBoozeApi from "./Api";

/**
 * This component upload drink data on the UI
 */
function DrinkItem() {
  const { id } = useParams();
  const [ drink, setDrink ] = useState({});

  useEffect(() => {

    const getDrink = async () => {
      try{
        const drink = await SnackOrBoozeApi.getDrink(id);
        setDrink(drink)
      }catch{
        setDrink(false);
      }
    }

    getDrink();

  },[]);

  if(!drink){
    return(<Redirect to={`/drinks`}/>)
  }
  
  return !drink.id? <p>Loading &hellip;</p>: (<section>
                                              <Card>
                                                <CardBody>
                                                  <CardTitle className="font-weight-bold text-center">
                                                    {drink.name}
                                                  </CardTitle>
                                                  <CardText className="font-italic">{drink.description}</CardText>
                                                  <p>
                                                    <b>Recipe:</b> {drink.recipe}
                                                  </p>
                                                  <p>
                                                    <b>Serve:</b> {drink.serve}
                                                  </p>
                                                  <p>
                                                    <Link to={`/edit/drinks/${drink.id}`}>Edit drink</Link>
                                                  </p>
                                                  <p>
                                                    <Link to="/drinks">Drinks menu</Link>
                                                  </p>
                                                </CardBody>
                                              </Card>
                                            </section>
                                          );

}

export default DrinkItem;