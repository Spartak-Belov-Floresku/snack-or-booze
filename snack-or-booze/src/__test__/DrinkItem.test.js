import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import DrinkItem from '../DrinkItem';

/** render DrinkItem component */
test('renders learn react link', async () => {
    render(
        <MemoryRouter initialEntries={["/drinks/martini/" ]}>
            <Route path={"/drinks/:id"} component={DrinkItem} />
        </MemoryRouter>
    );
    
    await screen.findByText("Martini")
});

/** Matching snapshot of DrinkItem components */
test('renders learn react link', async () => {
    
    const {asFragment} =  render(
        <MemoryRouter initialEntries={["/drinks/martini"]}>
            <Route path={"/drinks/:id"} component={DrinkItem} />
        </MemoryRouter>
    );

    await screen.findByText("Martini")
    expect(asFragment()).toMatchSnapshot(); 
    
});