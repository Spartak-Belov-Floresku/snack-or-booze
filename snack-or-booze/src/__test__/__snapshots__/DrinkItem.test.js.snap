// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <section>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Martini
        </h5>
        <p
          class="font-italic card-text"
        >
          An ice-cold, refreshing classic.
        </p>
        <p>
          <b>
            Recipe:
          </b>
           Mix 3 parts vodka & 1 part dry vermouth.
        </p>
        <p>
          <b>
            Serve:
          </b>
           Serve very cold, straight up.
        </p>
        <p>
          <a
            href="/edit/drinks/martini"
          >
            Edit drink
          </a>
        </p>
        <p>
          <a
            href="/drinks"
          >
            Drinks menu
          </a>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
